drivers/pwm/Kconfig.adi_max6639

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2025 Tenstorrent AI ULC
+# SPDX-License-Identifier: Apache-2.0
+
+config PWM_ADI_MAX6639
```

Comment:
I would change this to `PWM_MAX6639` to be consistent with what is upstream.

https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/pwm/Kconfig.max31790